pkg/server/v3/server_test.go

```diff
@@ -659,3 +659,99 @@ func TestCallbackError(t *testing.T) {
 		})
 	}
 }
+
+// BENCHMARKS =====================================================================================================
+
+func BenchmarkResponseHandlers(b *testing.B) {
+	for _, typ := range testTypes {
+		b.Run(typ, func(b *testing.B) {
+			config := makeMockConfigWatcher()
+			s := server.NewServer(context.Background(), config, server.CallbackFuncs{})
+
+			for n := 0; n < b.N; n++ {
+				// SOTW benchmarks fails if we don't set the responses inside the bench loop, not sure why
```

Comment:
Can you try to find out? Seems less than ideal to construct these as part of the benchmark loop

Reply:
Yup I'll look into this, if I can't figure anything out the worst we could to here is add a `//TODO` to investigate

Reply:
Ah okay I think I found the issue:
in the mock `CreateWatch`:
```go
func (config *mockConfigWatcher) CreateWatch(req *discovery.DiscoveryRequest, out chan cache.Response) func() {
	config.counts[req.TypeUrl] = config.counts[req.TypeUrl] + 1
	if len(config.responses[req.TypeUrl]) > 0 {
		out <- config.responses[req.TypeUrl][0]
		config.responses[req.TypeUrl] = config.responses[req.TypeUrl][1:]
	} else {
		config.watches += 1
		return func() {
			config.watches -= 1
		}
	}
	return nil
}
```
We seem to remove the response after we send it so the first round passes of the benchmark but everytime after it'll get no response. I logged out inside the loop: `b.Logf("TYP: %s \n \n %+v", typ, config.responses)` and it seems like they get empty after the first response is completed